decisions/2022-12-12_patient_and_program_transfer.md

```diff
@@ -0,0 +1,92 @@
+# Patient and program transfers
+
+- _Date_: 2022-12-12
+- _Deciders_:
+- _Status_:
+- _Outcome_:
+
+## Context
+
+Patients and programs document need to be accessible at multiple stores.
+This KDD tries to answer the following questions:
+
+- How to search, access and edit patient across multiple stores?
+- How to manage programs at multiple stores and transfer programs between stores?
+- How to manage program permission for different stores?
+
+**Summary of existing mSupply patient editing**:
+
+- Patients are only editable on the home store
+- Only patients from the `name` table can be edited (other names can't be edited). Note, patient push needs to be implemented in omSupply.
+- The patient's home store is identified in the `name` table by the `supplying_store_id` column
+- Its currently not directly supported to change a patient's home store
+
+**Summary of existing mSupply patient search and patient visibility**:
+
+- The endpoint `/patient-search` looks up all patients on central (no access control). The endpoint returns all patient information for a found patient.
+- The client can choose a patient and call `/nsj` to request a new `name_store_join` in order to make the patient visible in a store.
+- The client doesn't wait for sync but directly stores the fetched `name` and `name_store_join` records (client might receive entries again through sync)
+- patient is now visible on the store and syncs to the store (because of the `name_store_join`)
+
+**Transfer Programs Requirements**
+In general, a patient can have a list of programs and each program can be treated at different stores/clinic.
+A patient may change the clinic where the program is treated.
+Thus we need to be able to record the program store affiliation.
+Potentially a program can be treated at multiple stores(?).
+Its probably also useful to have a clinic transfer history.
+
+There are three types of transfers:
+
+- External in transfers: program transfer from an external clinic to mSupply
+- External out transfers: program is not manage by mSupply anymore
+- Internal transfers, a program is transferred to a different store/clinic within mSupply
+
+**Patient visit**
+A patient might have a program encounter at another store.
+To be clarified first:
+
+- user at visited store doesn't has access to the program data?
+- user can add a program encounter but not access other patient program and encounter data?
+
+## Options
+
+### Option 1
+
+**Patient editing:**
+
+- When changing a patient document the patient `name` is updated and pushed to central
+- For this to work we need to allow client to edit patients at stores that are not the patient's home store (?). (Because there is no fixed store affiliation for programs)
```

Comment:
From quite a few recent discussions, it's been a general consensus that this restriction should not be there:
* Information in the `name` records itself should be editable anywhere, as it's a very low probability of occurring and low impact (risk of discrepancy), since core patient details are not likely to change concurrently at two different sites.
* Just for reference, there is other data (mainly insurance related), that has higher probability and high impact if it's edited concurrently (the main reasons for home store only editability).